client_test.go

```diff
@@ -69,15 +78,63 @@ func ExampleNewClient() {
 	}
 
 	// use a devnull connector for example purposes
-	conn := &devnull.Connector{}
+	conn, err := dosaRenamed.GetConnector("devnull", nil)
+	if err != nil {
+		panic(err)
+	}
 
 	// create the client using the registry and connector
 	client := dosaRenamed.NewClient(reg, conn)
 
 	err = client.Initialize(context.Background())
 	if err != nil {
-		errors.Wrap(err, "client.Initialize returned an error")
+		panic(err)
+	}
+}
+
+// ExampleGetConnector gets an in-memory connector that can be used for testing your code
+// The in-memory connector always starts off with no rows, so you'll need to add rows to
+// your "database" before reading them
+func ExampleGetConnector() {
+	// register your entities so the engine can separate your data based on table names
+	// Scopes and prefixes are not used by the in-memory connector, and are ignored, but
+	// your list of entities is important. In this case, we only have one, our ClientTestEntity1
+	reg, err := dosaRenamed.NewRegistrar("test", "myteam.myservice", &ClientTestEntity1{})
+	if err != nil {
+		panic(err)
+	}
+
+	// Find the memory connector. There is no configuration information so pass a nil
+	// For this to work, you must force the init method of memory to run first, which happens
+	// when we imported memory in the import list, with an underscore to just get the side effects
```

Comment:
Can we "ear-mark" this for discussion? I don't want to shave any more yaks, but we should be consistent, meaning either provide constructor methods or use this paradigm consistently throughout all connectors.